subsys/bluetooth/rpc/host/bt_rpc_gatt_host.c

```diff
@@ -950,3 +1065,617 @@ static void bt_gatt_cb_register_handler(CborValue *value, void *handler_data)
+	container = get_subscribe_container(remote_pointer, &new_container);
+	if (container == NULL) {
+		ser_decoding_done_and_check(value);
+		goto alloc_error;
+	}
+	bt_gatt_subscribe_params_dec(value, &container->params);
+
+	if (!ser_decoding_done_and_check(value)) {
+		goto decoding_error;
+	}
+
+	result = bt_gatt_resubscribe(id, peer, &container->params);
+
+	ser_rsp_send_int(result);
+
+	if (result < 0 && new_container) {
+		free_subscribe_container(container);
+	}
+
+	return;
+decoding_error:
+	if (new_container) {
+		free_subscribe_container(container);
+	}
+alloc_error:
+	report_decoding_error(BT_GATT_RESUBSCRIBE_RPC_CMD, _handler_data);
+
+}
+
+NRF_RPC_CBOR_CMD_DECODER(bt_rpc_grp, bt_gatt_resubscribe, BT_GATT_RESUBSCRIBE_RPC_CMD,
+	bt_gatt_resubscribe_rpc_handler, NULL);
+
+static void bt_gatt_unsubscribe_rpc_handler(CborValue *value, void *_handler_data)
+{
+
+	struct bt_conn * conn;
+	int result;
+	uintptr_t remote_pointer;
+	struct bt_gatt_subscribe_container *container;
+	bool new_container = false;
+	
+	conn = bt_rpc_decode_bt_conn(value);
+	remote_pointer = ser_decode_uint(value);
+
+	if (!ser_decoding_done_and_check(value)) {
+		goto decoding_error;
+	}
+
+	container = get_subscribe_container(remote_pointer, &new_container);
+
+	if (container != NULL) {
+		result = bt_gatt_unsubscribe(conn, &container->params);
+		free_subscribe_container(container);
+	} else {
+		result = -EINVAL;
+	}
+
+	ser_rsp_send_int(result);
+
+	return;
+decoding_error:
+	report_decoding_error(BT_GATT_UNSUBSCRIBE_RPC_CMD, _handler_data);
+
+}
+
+NRF_RPC_CBOR_CMD_DECODER(bt_rpc_grp, bt_gatt_unsubscribe, BT_GATT_UNSUBSCRIBE_RPC_CMD,
+	bt_gatt_unsubscribe_rpc_handler, NULL);
+
+static void bt_rpc_gatt_subscribe_flag_update_rpc_handler(CborValue *value, void *_handler_data)
+{
+	uint32_t flags_bit;
+	int result;
+	uintptr_t remote_pointer;
+	struct bt_gatt_subscribe_container *container;
+	bool new_container = false;
+	int val;
+
+	remote_pointer = ser_decode_uint(value);
+	flags_bit = ser_decode_uint(value);
+	val = ser_decode_int(value);
+
+	if (!ser_decoding_done_and_check(value)) {
+		goto decoding_error;
+	}
+
+	container = get_subscribe_container(remote_pointer, &new_container);
+	if (container == NULL) {
+		result = -EINVAL;
+	} else {
+		if (atomic_test_bit(container->params.flags, flags_bit)) {
+			result = 1;
+		} else {
+			result = 0;
+		}
+
+		if (val == 0) {
+			atomic_clear_bit(container->params.flags, flags_bit);
+		} else if (val > 0) {
+			atomic_set_bit(container->params.flags, flags_bit);
+		}
+	}
+
+	ser_rsp_send_int(result);
+
+	return;
+decoding_error:
+	report_decoding_error(BT_RPC_GATT_SUBSCRIBE_FLAG_UPDATE_RPC_CMD, _handler_data);
+
+}
+
+NRF_RPC_CBOR_CMD_DECODER(bt_rpc_grp, bt_rpc_gatt_subscribe_flag_update, BT_RPC_GATT_SUBSCRIBE_FLAG_UPDATE_RPC_CMD,
+	bt_rpc_gatt_subscribe_flag_update_rpc_handler, NULL);
+
+int bt_rpc_gatt_subscribe_flag_set(struct bt_gatt_subscribe_params *params, uint32_t flags_bit)
```

Comment:
What is a use case ?

Reply:
I added also to RPC host to make the API the same on both sides.